core/bin/elasticsearch/create_index.rs

```diff
@@ -0,0 +1,152 @@
+use std::collections::HashMap;
+
+use clap::{Parser, ValueEnum};
+use elasticsearch::auth::Credentials;
+use elasticsearch::http::transport::{SingleNodeConnectionPool, TransportBuilder};
+use elasticsearch::indices::{IndicesCreateParts, IndicesExistsParts};
+use elasticsearch::Elasticsearch;
+use http::StatusCode;
+use url::Url;
+
+#[derive(Parser, Debug, Clone, ValueEnum)]
+enum Region {
+    Local,
+    #[clap(name = "us-central-1")]
+    UsCentral1,
+}
+
+impl std::fmt::Display for Region {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Region::Local => write!(f, "local"),
+            Region::UsCentral1 => write!(f, "us-central-1"),
+        }
+    }
+}
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    #[arg(long, help = "The index name (without the version)")]
+    index_name: String,
+
+    #[arg(long, help = "The version of the index")]
+    index_version: u32,
+
+    #[arg(long, help = "The region of the index (local or us-central-1)")]
+    #[arg(value_enum)]
+    region: Region,
```

Comment:
Ideally should be pulled from the env variable, failing if not set?

Reply:
That's what I had initially done
I changed it because for index creation likely run in prodbox, explicit seemed better (for running purposes using the env var is the intent)

I coul change again to fetch from env var of param not set, and ask for confirmation in that case?

Reply:
I would personally fetch from env var and always ask for confirmation to minimize typo/errors but that's not super important. Any approach is fine.

Reply:
changed with DUST_REGION and a skip-confirmation arg for scripts

Reply:
yes; I updated the runbook for search infra and will also update local setup runbook
we use DUST_REGION=local